### PR TITLE
Added endpoint for JWT creation

### LINKED ISSUE #28: Error handling

### PR DESCRIPTION
**Pull request by mperovicmrsovic**
Added endpoint for jwt creation. Work in progress for #1.
Related with pr #29.

In commit _690a927_ I created endpoint for jwt token creation ( _**api/users/auth**_ ) .
User Admin is set to create on applications start, for now (_email_: "**fmadmin@vegait.rs**", _password_: "**foodAdmin**"). 
If credentials that user provides during login are wrong I just put to throw new runtime exception with message "Wrong credentials". **It should be changed after we add global exception handling**.
 
Exception handling is done in commit [c586c61](https://github.com/food-manager/backend/pull/30/commits/c586c61eb75b510823c37526e24e9b82d58dc7b5), closes #28

### CODE REVIEW

File: src/main/java/com/g1dra/foodmanager/FoodManagerApplication.java [hunk 1]
```diff
@@ -10,4 +18,22 @@ public static void main(String[] args) {
...
+                                .name("Admin")
+                                .email("fmadmin@vegait.rs")
+                                .password(hashedPassword)
+                                .role(UserRole.ADMIN)
+                                .createdAt(LocalDateTime.now())
+                                .build()
+                );
+            }
+        };
+    }
```
Comment (d-vucetic) [inline src/main/java/com/g1dra/foodmanager/FoodManagerApplication.java:38]: Extract to separate package, lets say we name it seeders. Then there we can have. InitSeederClass

File: src/main/java/com/g1dra/foodmanager/config/JwtRequest.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.sun.istack.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import javax.validation.constraints.Size;
+
+@Data
+@AllArgsConstructor
+public class JwtRequest {
```
Comment (d-vucetic) [inline src/main/java/com/g1dra/foodmanager/config/JwtRequest.java:11]: I'm not sure about name of this class. We name is describing response that will get(a jwt). So name of this class should be LoginRequest or similar.

**Review by d-vucetic — CHANGES_REQUESTED** [review]

**Comment by d-vucetic** [comment]
Also pls add test to check if jwt is created. Good method to use when writing tests is: **[Given-When-Then](https://en.wikipedia.org/wiki/Given-When-Then#:~:text=Given%2DWhen%2DThen%20(GWT,words%20given%2C%20when%20and%20then.)**

- given: we have a request with username and password
- when: we make request on auth endpint
- then: we except a response with JWT token

File: src/main/java/com/g1dra/foodmanager/FoodManagerApplication.java (hunk 1 above)
Comment (mperovicmrsovic) [reply src/main/java/com/g1dra/foodmanager/FoodManagerApplication.java]: I added seeder package and class for initial data in commit [38575fb](https://github.com/food-manager/backend/pull/30/commits/38575fbb44fbf707f54c3b572a6472a9f241e706)

File: src/main/java/com/g1dra/foodmanager/config/JwtRequest.java (hunk 2 above)
Comment (mperovicmrsovic) [reply src/main/java/com/g1dra/foodmanager/config/JwtRequest.java]: Renamed into AuthRequest, commit [1769c0a](https://github.com/food-manager/backend/pull/30/commits/1769c0ad2478c33638272684d0fe7ca666b18dd5)

**Comment by d-vucetic** [comment]
Tests required now..

**Review by d-vucetic — APPROVED** [review]

**Pull request merged**